src/app/clusters/general_diagnostics_server/general_diagnostics_server.cpp

```diff
@@ -107,26 +108,82 @@ CHIP_ERROR GeneralDiagosticsAttrAccess::Read(const ConcreteAttributePath & aPath
         return ReadNetworkInterfaces(aEncoder);
     }
     case RebootCount::Id: {
-        return ReadIfSupported(&PlatformManager::GetRebootCount, aEncoder);
+        return ReadIfSupported(&DeviceLayer::PlatformManager::GetRebootCount, aEncoder);
     }
     case UpTime::Id: {
-        return ReadIfSupported(&PlatformManager::GetUpTime, aEncoder);
+        return ReadIfSupported(&DeviceLayer::PlatformManager::GetUpTime, aEncoder);
     }
     case TotalOperationalHours::Id: {
-        return ReadIfSupported(&PlatformManager::GetTotalOperationalHours, aEncoder);
+        return ReadIfSupported(&DeviceLayer::PlatformManager::GetTotalOperationalHours, aEncoder);
```

Comment:
We never mark UpTime and TotalOperationalHours dirty.  Shouldn't we be doing that?  Or at least have a followup tracking it?

Reply:
Spec say those two attributes should not be subscribed, and those two attributes should only be read 
"A Node interested in the value of the UpTime attribute SHOULD NOT subscribe to value changes but SHOULD perform a single read of the value."

Reply:
Should not doesn't mean you can't do it, just that it's not necessarily a great idea.

If someone _does_ subscribe to them, the spec as currently written requires reports to be sent on changes.

Now I think this is silly and we should change the spec, personally, but that needs an SDK feedback spec issue, etc.

Reply:
most of attributes in diagnostic clusters suffer the same problem, since the values of some diagnostic clusters are consistently changing. Subscribe to those attributes will flood the reporting

Reply:
I don't disagree.  I am just pointing out that if we think that our impl now follows the spec we are wrong: it does not.  If the spec is requiring things that make no sense to implement, we should be filing spec issues, not just silently ignoring what it says and doing something else.

Reply:
I filed https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/4612